token-swap/program/fuzz/fuzz_targets/instructions.rs

```diff
@@ -0,0 +1,258 @@
+#![no_main]
+
+use spl_token_swap_fuzz::{get_token_balance, AccountData, TokenSwapAccountInfo};
+
+use spl_token_swap::{
+    curve::{
+        base::{CurveType, SwapCurve},
+        constant_product::ConstantProductCurve,
+    },
+    error::SwapError,
+    instruction::{Deposit, Swap, Withdraw},
+};
+
+use spl_token::error::TokenError;
+
+use arbitrary::Arbitrary;
+use libfuzzer_sys::fuzz_target;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug, Arbitrary, Clone)]
+enum FuzzInstruction {
+    Swap {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        trade_direction: TradeDirection,
+        instruction: Swap,
+    },
+    Deposit {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        pool_token_id: AccountId,
+        instruction: Deposit,
+    },
+    Withdraw {
+        token_a_id: AccountId,
+        token_b_id: AccountId,
+        pool_token_id: AccountId,
+        instruction: Withdraw,
+    },
+}
+
+/// Helper enum to tell which direction a swap is meant to go.
+#[derive(Debug, Arbitrary, Clone)]
+enum TradeDirection {
+    AtoB,
+    BtoA,
+}
+
+/// Use u8 as an account id to simplify the address space and re-use accounts
+/// more often.
+type AccountId = u8;
+
+const INITIAL_SWAP_TOKEN_A_AMOUNT: u64 = 100_000_000_000;
+const INITIAL_SWAP_TOKEN_B_AMOUNT: u64 = 300_000_000_000;
+
+const INITIAL_USER_TOKEN_A_AMOUNT: u64 = 1_000_000_000;
+const INITIAL_USER_TOKEN_B_AMOUNT: u64 = 3_000_000_000;
+
+fuzz_target!(|fuzz_instructions: Vec<FuzzInstruction>| {
+    run_fuzz_instructions(fuzz_instructions)
+});
+
+fn run_fuzz_instructions(fuzz_instructions: Vec<FuzzInstruction>) {
+    let trade_fee_numerator = 25;
+    let trade_fee_denominator = 10000;
+    let owner_trade_fee_numerator = 5;
+    let owner_trade_fee_denominator = 10000;
+    let owner_withdraw_fee_numerator = 30;
+    let owner_withdraw_fee_denominator = 10000;
+    let host_fee_numerator = 1;
+    let host_fee_denominator = 5;
+    let swap_curve = SwapCurve {
+        curve_type: CurveType::ConstantProduct,
+        calculator: Box::new(ConstantProductCurve {
+            trade_fee_numerator,
+            trade_fee_denominator,
+            owner_trade_fee_numerator,
+            owner_trade_fee_denominator,
+            owner_withdraw_fee_numerator,
+            owner_withdraw_fee_denominator,
+            host_fee_numerator,
+            host_fee_denominator,
+        }),
+    };
```

Comment:
We can eventually add more curves or even fuzz the curves, but that would be an initialize test as well